src/Confluent.Kafka.Avro/AvroDeserializer.cs

```diff
@@ -173,24 +173,19 @@ public T Deserialize(string topic, byte[] array)
                 var writerIdBigEndian = reader.ReadInt32();
                 var writerId = IPAddress.NetworkToHostOrder(writerIdBigEndian);
 
-                DatumReader<T> datumReader = null;
-
-                lock (datumReaderLock)
+                datumReaderBySchemaId.TryGetValue(writerId, out var datumReader);
+                if (datumReader == null)
                 {
-                    datumReaderBySchemaId.TryGetValue(writerId, out datumReader);
-                    if (datumReader == null)
-                    {
-                        // it's a good idea to retain the lock over this blocking call since there
-                        // may be concurrent deserialize calls for the same T, and in that case it's
-                        // best not to hit Schema Registry more than once for the same information.
-                        var writerSchemaJson = SchemaRegistryClient.GetSchemaAsync(writerId).Result;
-                        var writerSchema = Avro.Schema.Parse(writerSchemaJson);
-
-                        datumReader = new SpecificReader<T>(writerSchema, ReaderSchema);
-                        datumReaderBySchemaId[writerId] = datumReader;
-                    }
+                    // it's a good idea to retain the lock over this blocking call since there
+                    // may be concurrent deserialize calls for the same T, and in that case it's
+                    // best not to hit Schema Registry more than once for the same information.
+                    var writerSchemaJson = SchemaRegistryClient.GetSchemaAsync(writerId).ConfigureAwait(false).GetAwaiter().GetResult();
+                    var writerSchema = Avro.Schema.Parse(writerSchemaJson);
+
+                    datumReader = new SpecificReader<T>(writerSchema, ReaderSchema);
+                    datumReaderBySchemaId[writerId] = datumReader;
```

Comment:
There is a race condition here between TryGetValue and the insert, the original lock is needed (which makes the use of Concurrent dict excessive).

Reply:
Good point but I was thinking that the race condition wouldn't matter that much as it just means there is a single additional call to the schema registry.

The more important part of the change,  is the inclusion of the .ConfigureAwait(false) as that helps prevent the deadlocks caused by context captures.

Reply:
I added `ConfigureAwait(false)` to all SR method calls.